src/notify/app/queries/consumer.py

```diff
@@ -0,0 +1,29 @@
+from functools import lru_cache
+from sqlalchemy.orm import Session
+from app.models import Consumer
+from app.utils.database import engine
+
+
+def fetch_from_cache(key: str) -> Consumer | None:
+    return indexed_by_key().get(key)
+
+
+def indexed_by_key() -> dict[str, Consumer]:
+    index = {}
+
+    for c in fetch_all_cached():
+        index[c.key] = c
+
+    return index
+
+
+@lru_cache(maxsize=1)
```

Comment:
not sure i understand the `maxsize` argument, will setting it as 1 mean the cache invalidates after 1 call?